src/MSBuild/XMake.cs

```diff
@@ -1725,6 +1725,12 @@ internal static void GatherCommandLineSwitches(ArrayList commandLineArgs, Comman
                                 String.Equals(switchName, "maxcpucount", StringComparison.OrdinalIgnoreCase))
                             {
                                 int numberOfCpus = Environment.ProcessorCount;
+                                if (IntPtr.Size == 4 && NativeMethodsShared.IsWindows) // 32-bit process, 32-bit Windows had a 32-core limit
```

Comment:
Is there a more intuitive way to get that it's 32-bit? If not, this is fine. I was only confused until I read the comment.

Reply:
If there isn't another way, can we comment with a link to https://docs.microsoft.com/en-us/dotnet/api/system.intptr.size?view=netcore-3.1, or the excerpt from that page
`The size of a pointer or handle in this process, measured in bytes. The value of this property is 4 in a 32-bit process, and 8 in a 64-bit process. You can define the process type by setting the /platform switch when you compile your code with the C# and Visual Basic compilers.`

Reply:
To be clear, I understood immediately after reading the comment. Although I'd count this as somewhat subtle but definitely simple and functional.

Reply:
Maybe extract it to a readonly static property is32bit? Or use an enum?

Reply:
Good point! You reminded me of https://github.com/dotnet/msbuild/blob/master/src/Shared/EnvironmentUtilities.cs#L11

We should follow this model.

Reply:
Just use this but test for the negation?

Reply:
Also it looks like there is a static bool in environment.
https://github.com/dotnet/runtime/blob/ea2b09beea535abf960651c59359e40f99f2fecf/src/libraries/System.Private.CoreLib/src/System/Environment.cs#L138

Reply:
Either way works, but I'd make a new `Is32Bit` version. We probably don't have to worry about 16-bit machines, and we don't currently have to worry about 128-bit machines, but just in case this code lasts a while...

Reply:
`EnvironmentUtilities` is internal to this assembly thus cannot be referenced without adding an InternalsVisibleTo which I don't think we want to do. Used `!Environment.Is64BitProcess` instead.